File: backend/app/models/deployment_log.py
from typing import Optional
from sqlalchemy import String, Integer, Text, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship

from .base import Base
from .token_configuration import TokenConfiguration


class DeploymentLog(Base):
    """Model for tracking token deployment status across different chains"""
    
    __tablename__ = "deployment_logs"

    # Deployment details
    token_config_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("token_configurations.id", ondelete="CASCADE"),
        nullable=False
    )
    chain_name: Mapped[str] = mapped_column(String, nullable=False)
    chain_id: Mapped[str] = mapped_column(String, nullable=False)
    contract_address: Mapped[Optional[str]] = mapped_column(String, nullable=True)
    status: Mapped[str] = mapped_column(
        String,
        nullable=True,
        server_default="pending"
    )
    error_message: Mapped[Optional[str]] = mapped_column(Text, nullable=True)

    # Relationship
    token_config: Mapped[TokenConfiguration] = relationship(
        "TokenConfiguration",
        back_populates="deployment_logs"
    ) 

File: backend/src/models/DeploymentLog.js
const { DataTypes } = require('sequelize');
const sequelize = require('../db/config');
const TokenConfiguration = require('./TokenConfiguration');

const DeploymentLog = sequelize.define('DeploymentLog', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  tokenConfigId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    field: 'token_config_id',
    references: {
      model: 'token_configurations',
      key: 'id'
    }
  },
  chainName: {
    type: DataTypes.STRING,
    allowNull: false,
    field: 'chain_name'
  },
  chainId: {
    type: DataTypes.STRING,
    allowNull: false,
    field: 'chain_id'
  },
  contractAddress: {
    type: DataTypes.STRING,
    field: 'contract_address'
  },
  status: {
    type: DataTypes.STRING,
    allowNull: false,
    defaultValue: 'pending'
  },
  transactionHash: {
    type: DataTypes.STRING,
    field: 'transaction_hash'
  },
  errorMessage: {
    type: DataTypes.TEXT,
    field: 'error_message'
  }
}, {
  tableName: 'deployment_logs',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at'
});

// Establish relationship
DeploymentLog.belongsTo(TokenConfiguration, { foreignKey: 'tokenConfigId' });
TokenConfiguration.hasMany(DeploymentLog, { foreignKey: 'tokenConfigId' });

module.exports = DeploymentLog; 